event_model/documents/event_descriptor.py

```diff
@@ -0,0 +1,121 @@
+from typing import Any, Dict, List
+
+from typing_extensions import Annotated, NotRequired, Literal, TypedDict
+
+from .generate.type_wrapper import Field, AsRef
+
+
+class DataKey(TypedDict):
+    external: NotRequired[
```

Comment:
Can we have docstrings for these intermediate classes where they exist in the existing json schemas please?

Reply:
Bugs ironed out in the latest commit, now all sub-typed dicts (those referenced in the annotations of an attribute in a typeddict) are swapped out for generated basemodels.